File: client/src/types/JsonModels/ProgramPlans/ProgramScheduleRuleRequest.ts
import type { DayOfWeek } from "../Enums/DayOfWeek";
import type { ProgramPlanDayType } from "../Enums/ProgramPlanDayType";

export interface ProgramScheduleRuleRequest
{
	dayOfWeek?: DayOfWeek;
	rotationDayIndex?: number;
	dayType: ProgramPlanDayType;
	workoutTemplateId?: number;
	weekInterval: number;
	orderIndex: number;
	isOptional: boolean;
}

File: server/FitMate.Core/JsonModels/ProgramPlans/ProgramScheduleRuleRequest.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.ProgramPlans;

public class ProgramScheduleRuleRequest
{
    public DayOfWeek? DayOfWeek { get; set; }
    public int? RotationDayIndex { get; set; }
    public ProgramPlanDayType DayType { get; set; }
    public long? WorkoutTemplateId { get; set; }
    public int WeekInterval { get; set; } = 1;
    public int OrderIndex { get; set; }
    public bool IsOptional { get; set; }
}
